### DISCUSSION TITLE
How to add a folder on `pages` without adding a new path in the app route?

### DISCUSSION THREAD

**Discussion by mauriciosoares** [vercel/next.js#30588] [Help]
I want to start using the new middlewares feature to identify if my user is logged in or not in my app, and redirect based on that... So I wanted to have a middleware that would run only on the `/sign-in` and `/sign-up` routes, and if my user is already logged in, I redirect them to the app route. However I don't want my `/` (index) page to be covered by that middleware. Essentially:

```
https://myapp.com/sign-in -> should execute the middleware
https://myapp.com/sign-up -> should execute the middleware
https://myapp.com/ -> should NOT execute the middleware
```

The problem is that all of these files are in the root of my `pages` folder, and if I add a `_middleware.ts` file in there, my `index` route will be affected by it to... So is there any way to add a folder that wouldn't force me to add an additional path to the `sign-in/up` url? A pseudo example would look something like this:

```
pages/
├─ []/
│  ├─ sign-in.tsx
│  ├─ sign-up.tsx
│  ├─ _middleware.ts
├─ index.tsx
```

The middleware in this case would only run for `sign-in/up`, but it would still keep these 2 paths as `myapp.com/sign-in/up`

**Comment by gaston-flores** [comment]
Maybe you can use [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) to hide the extra folder:
```
pages/
├─ auth/
│  ├─ sign-in.tsx
│  ├─ sign-up.tsx
│  ├─ _middleware.ts
├─ index.tsx
```
```js
// next.config.js
module.exports = {
  async rewrites() {
    return [
      {
        source: '/sign-in',
        destination: '/auth/sign-in',
      },
      {
        source: '/sign-up',
        destination: '/auth/sign-up',
      },
    ]
  },
}
```

**Reply by mauriciosoares** [reply]
Oh, didn't know about `rewrites` @gaston-flores, that's cool, and will certainly do the job

Thanks man. I'll keep this discussion opened for just a bit longer to see if there are any different approaches I can take, but your solution will for sure help me out.

**Reply by mauriciosoares** [reply]
Well, actually looks like when I use the rewrites option, and I enter the page directly on `myapp.com/sign-in`, the middleware inside the `auth` folder does not get executed...

I wonder if this is intentional by design, or possibly a bug... 🤔
